src/ooo_cpu.cc

```diff
@@ -441,8 +491,8 @@ void O3_CPU::fetch_instruction()
     }
 
     // send to DECODE stage
-    std::size_t checked_for_decode = 0;
-    while (checked_for_decode < DECODE_WIDTH && IFETCH_BUFFER.occupancy > 0 && DECODE_BUFFER.occupancy < DECODE_BUFFER.SIZE &&
+    available_fetch_bandwidth = FETCH_WIDTH;
+    while (available_fetch_bandwidth > 0 && IFETCH_BUFFER.occupancy > 0 && DECODE_BUFFER.occupancy < DECODE_BUFFER.SIZE &&
```

Comment:
This change appears to have no effect except to make a counter that goes down instead of a counter that goes up. Why is that?

Reply:
Correct, that is the only functional difference, but I believe it improves readability as it makes it more clear that the following loop is going to consume the FETCH_WIDTH resource.  We start with a full stock of the resource, and  as we go through the loop, the resource is drained, and when we run out, the loop is exited.  This matches the code pattern of how cache bandwidth is shared between loads and prefetches in cache.cc.